documentation/new-version-guideline.md

```diff
@@ -0,0 +1,302 @@
+```
+### Reference
+
+For more information, you could refer to [design guidelines of LRO](https://azure.github.io/azure-sdk/golang_introduction.html#methods-invoking-long-running-operations) and [API reference of poller](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#Poller).
+
+## Client Options
+
+### Request Retry Policy
+The SDK provides a baked in retry policy for failed requests with default values that can be configured by `arm.ClientOptions.Retry`. For example:
+
+```go
+rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential,
+    &arm.ClientOptions{
+        ClientOptions: policy.ClientOptions{
+            Retry: policy.RetryOptions{
+                // retry for 5 times
+                MaxRetries: 5,
+            },
+        },
+    },
+)
+```
+
+### Customized Policy
+
+You can use `arm.ClientOptions.PerCallPolicies` and `arm.ClientOptions.PerRetryPolicies` option to inject customized policy to the pipeline. You can refer to `azcore` [document](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore) for further information.
+
+### Custom HTTP Client
+
+You can use `arm.ClientOptions.Transport` to set your own implementation of HTTP client. The HTTP client must implement the `policy.Transporter` interface. For example:
+
+```go
+// your own implementation of HTTP client
+httpClient := NewYourOwnHTTPClient{}
+rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential,
+    &arm.ClientOptions{
+        ClientOptions: policy.ClientOptions{
+            Transport: &httpClient,
+        },
+    },
+)
+```
+
+### Reference
+
+More client options can be found [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/policy/policy.go).
+
+
+## Troubleshooting
+
+### Logging
+
+The SDK uses the classification-based logging implementation in `azcore`. To enable console logging for all SDK modules, set AZURE_SDK_GO_LOGGING to all. 
+
+Use `LogOption` to configure log behavior. For example:
+```go
+rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential,
+    &arm.ClientOptions{
+        ClientOptions: policy.ClientOptions{
+            Logging: policy.LogOptions{
+                // include HTTP body for log
+                IncludeBody: true,
+            },
+        },
+    },
+)
+```
+
+Use the `azcore/log` package to control log event and write log to the desired location. For example:
+
+```go
+import azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+
+// print log output to stdout
+azlog.SetListener(func(event azlog.Event, s string) {
+    fmt.Println(s)
+})
+
+// include only azidentity credential logs
+azlog.SetEvents(azidentity.EventAuthentication)
+```
+
+### Raw HTTP response
+When there is an error in the SDK request, you need to convert the error to the `azcore.ResponseError` interface to get the raw HTTP response. When the request is successful, you can get the raw HTTP response from request context.
```

Comment:
Can we get the raw HTTP response from the context if there is an error?
The document here reads the answer is no

Reply:
I've changed the description. For error, you could either get from context or get from error.

Reply:
Therefore I believe we could provide one more snippet to get the raw response from ctx only